scipy/optimize/_minimize_constrained.py

```diff
@@ -0,0 +1,485 @@
+    method {"eq_constrained_sqp", "tr_interior_point"}
+        Optimization method used.
+    constr_violation : float
+        Constraint violation at last iteration.
+    optimality : float
+        Norm of the lagrangian gradient at last iteration.
+    fun : float
+        For the `eq_constraineq_sqp` method this is the objective function
+        evaluated at the solution and for the `tr_interior_point` method
+        this is the barrier  function evaluated at the solution.
+    grad : ndarray, (n,)
+        For the `eq_constraineq_sqp` method this is the gradient of the
+        objective function evaluated at the solution and for the
+        `tr_interior_point` method  this is the gradient of the barrier
+        function evaluated at the solution.
+    constr : ndarray, shape (n_ineq + n_eq,)
+        For the `eq_constraineq_sqp` method this is the equality constraint
+        evaluated at the solution and for the `tr_interior_point` method
+        this is he barrier function constraint.
+    jac : {ndarray, sparse matrix}, shape (n_ineq + n_eq, n)
+        For the `eq_constraineq_sqp` method this is the Jacobian matrix of
+        the equality constraint evaluated at the solution and for the
+        `tr_interior_point` method his is Jacobian matrix of the barrier
+        function constraint.
+
+    Notes
+    -----
+    Method `equality_constrained_sqp` is an implementation of
+    Byrd-Omojokun Trust-Region SQP method described [3]_ and
+    in [2]_, p. 549. It solves equality constrained equality
+    constrained optimization problems by solving, at each substep,
+    a trust-region QP subproblem. The inexact solution of these
+    QP problems using projected CG method makes this method
+    appropriate for large-scale problems.
+
+    Method `tr_interior_point` is an implementation of the
+    trust-region interior point method described in [1]_.
+    It solves general nonlinear by introducing slack variables
+    and solving a sequence of equality-constrained barrier problems
+    for progressively smaller values of the barrier parameter.
+    The previously described equality constrained SQP method is used
+    to solve the subproblems with increasing levels of accuracy as
+    the iterate gets closer to a solution. It is also an
+    appropriate method for large-scale problems.
+
+    References
+    ----------
+    .. [1] Byrd, Richard H., Mary E. Hribar, and Jorge Nocedal.
+           "An interior point algorithm for large-scale nonlinear
+           programming." SIAM Journal on Optimization 9.4 (1999): 877-900.
+    .. [2] Nocedal, Jorge, and Stephen J. Wright. "Numerical optimization"
+           Second Edition (2006).
+    .. [3] Lalee, Marucha, Jorge Nocedal, and Todd Plantenga. "On the
+           implementation of an algorithm for large-scale equality
+           constrained optimization." SIAM Journal on
+           Optimization 8.3 (1998): 682-706.
+    """
+    # Put ``constraints`` in list format
```

Comment:
Definitely could use some input validation/cleanup somewhere here that puts inputs in the preferred form (e.g. NumPy array rather than list) and checks for inconsistencies (e.g. in shapes) and unusual cases (e.g. empty array), raising intelligible exceptions as appropriate.